src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs

```diff
@@ -299,6 +299,39 @@ public static IEnumerable<object[]> Match_Basic_TestData()
 
             // Surrogate pairs splitted up into UTF-16 code units.
             yield return new object[] { @"(\uD82F[\uDCA0-\uDCA3])", "\uD82F\uDCA2", RegexOptions.CultureInvariant, 0, 2, true, "\uD82F\uDCA2" };
+
+            // AnyNewLine (with none of the special characters used as line ending)
+            yield return new object[] { @"line3\nline4$", "line1\nline2\nline3\nline4", RegexOptions.AnyNewLine, 0, 23, true, "line3\nline4" };
```

Comment:
Should we add tests with `\z` and it's combinations here?

Reply:
I didn't look at the PR, but one way or another we sohuld have tests that cover the whole of my table. It might be good to include it as a comment even given the amount of noodling we did about it.

https://github.com/dotnet/corefx/pull/41195#issuecomment-549623138

Subsequent to this PR, one of us should put a PR up against the API docs to include something like this table.

Reply:
I suggested in https://github.com/dotnet/corefx/issues/28410#issue-308079007 that the API Docs use the phrase "Anchor Modifier" to describe how `AnyNewLine` works, as well as incorporated the full table at the bottom of that comment.

Related, I had put together a table from Jeffrey Freidl's Mastering Regular Expressions to demonstrate how he explains these to users: https://github.com/jzabroski/Home/tree/master/RegularExpressions - Note, I like the way Jeffrey uses the phrase Concern and puts use cases in Plain English, but I actually think parts of his table are somewhat incorrect in trying to build a matrix of tests across all languages.